src/memory_server.ml

```diff
@@ -278,22 +298,51 @@ let calculate_boot_time_host_free_memory () =
       pages_to_kib (Int64.of_nativeint boot_time_host_free_pages) in
     Int64.mul 1024L boot_time_host_free_kib
 
+let calc_constant_boot_time_host_free_memory constant_count interval =
+  debug "Check boot time host free memory: constant-count=%d check-interval=%f(seconds)" constant_count interval;
+  let rec calc_constant existing_results =
+    let free_memory = calculate_boot_time_host_free_memory () in
+    debug "Calculating boot time host free memory: %Ld" free_memory;
+    if (List.length existing_results) = (constant_count - 1) then
+      if List.filter_map (fun x -> if x <> free_memory then Some(true) else None) existing_results = [] then begin
```

Comment:
This looks complicated to me. Would this be a case for `List.for_all` or `List.exists`. Taking `List.length` suggests you are  either using the wrong data structure or should be testing against an empty list.

Reply:
`List.exists` is more suitable in this case.

> Taking List.length suggests you are either using the wrong data structure or should be testing against an empty list.
Why it should be testing against an empty list?

Reply:
What I tried to say: testing against the lengths of a list often suggests it is the wrong data structure. Tests should be against the head of a list or against an empty list - in general, not here specifically.

Reply:
Correct! Your intuition is right!

Reply:
A value with the last amount of free memory and another with the time we've encountered it should be enough for this algorithm.

Reply:
Yes, that's what I am going to change the code.